### PR TITLE
Add ClusterRole to get, list and watch pullsubscriptions and storages

### LINKED ISSUE #252: Add ClusterRole to get, list and watch "pullsubscriptions" and "storages"
In PR https://github.com/knative/eventing/pull/1849, we added ClusterRole to get, list and watch "pullsubscriptions" and "storages" in config/200-source-resolvers-clusterrole.yaml.
This is a temporary solution. Such ClusterRole should eventually be moved to config directories in https://github.com/google/knative-gcp.

### PR DESCRIPTION
**Pull request by capri-xiyue**
Fixes #252  

## Proposed Changes

  * Add ClusterRole to get, list and watch pullsubscriptions and storages
  *
  *

**Release Note**

```release-note
None
```

### CODE REVIEW

**Comment by n3wscott** [comment]
/ok-to-test
/lgtm
/approve
/hold
hold until https://github.com/knative/eventing/pull/1856 lands

File: config/201-clusterrole.yaml [hunk 1]
```diff
@@ -132,3 +132,41 @@ rules:
...
+
+# The role is needed for the aggregated role source-resolver in knative-eventing to provide readonly access to "Sources"
+---
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: knative-gcp-pullsubscription-resolver
+  labels:
+    eventing.knative.dev/release: devel
+    duck.knative.dev/sourceObserver: "true"
```
Comment (n3wscott) [inline config/201-clusterrole.yaml]: this should be ` duck.knative.dev/sourceObserver`

File: config/201-clusterrole.yaml [hunk 2]
```diff
@@ -132,3 +132,41 @@ rules:
   resources:
     - eventtypes
   verbs: *everything
+
+# The role is needed for the aggregated role source-observer in knative-eventing to provide readonly access to "Sources"
```
Comment (Harwayne) [inline config/201-clusterrole.yaml]: The comment should be after the `---`.

```suggestion
# The role is needed for the aggregated role source-observer in knative-eventing to provide readonly access to "Sources".
# See https://github.com/knative/eventing/blob/master/config/200-source-observer-clusterrole.yaml.
```

File: config/201-clusterrole.yaml [hunk 3]
```diff
@@ -132,3 +132,41 @@ rules:
...
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: knative-gcp-pullsubscription-observer
+  labels:
+    eventing.knative.dev/release: devel
+    duck.knative.dev/source: "true"
+rules:
+  - apiGroups:
+      - "pubsub.cloud.run"
```
Comment (Harwayne) [inline config/201-clusterrole.yaml]: I think this changed in #332.

File: config/201-clusterrole.yaml [hunk 4]
```diff
@@ -132,3 +132,41 @@ rules:
...
+    resources:
+      - "pullsubscriptions"
+    verbs:
+      - get
+      - list
+      - watch
+
+# The role is needed for the aggregated role source-observer in knative-eventing to provide readonly access to "Sources"
```
Comment (Harwayne) [inline config/201-clusterrole.yaml]: Same comment as on the previous comment.

File: config/201-clusterrole.yaml [hunk 5]
```diff
@@ -132,3 +132,41 @@ rules:
...
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: knative-gcp-storage-observer
+  labels:
+    eventing.knative.dev/release: devel
+    duck.knative.dev/source: "true"
+rules:
+  - apiGroups:
+      - "events.cloud.run"
```
Comment (Harwayne) [inline config/201-clusterrole.yaml]: I think this changed in #332.

File: config/201-clusterrole.yaml (hunk 3 above)
Comment (capri-xiyue) [reply config/201-clusterrole.yaml]: This clusterrole never gets merged into master branch. Therefore Nacho can't see it and he didn't change it in #332

File: config/201-clusterrole.yaml (hunk 5 above)
Comment (capri-xiyue) [reply config/201-clusterrole.yaml]: This clusterrole never gets merged into master branch. Therefore Nacho can't see it and he didn't change it in #332

**Pull request merged**